### ISSUE TITLE
generating augmanted images

### ISSUE THREAD

**Issue by Harhiem** [rstudio/keras#156]
The fix of yesterday removed the error message, 

But unfortunately no augmented images are geneated from a single image and stored in the indicated directory. The only ouput in the console is   <keras.preprocessing.image.NumpyArrayIterator> 

library(keras)

img_width<-150
img_height<-150

img<-image_load("D:/data_flow17/jpg/image_0007.jpg",target_size=c(img_width,img_height))  
img1<-image_to_array(img) 
dim(img1)<-c(1,dim(img1))  
dim(img1)
gen_images <- image_data_generator(rotation_range = 20,
                                   width_shift_range = 0.30,
                                   height_shift_range = 0.30,
                                   horizontal_flip = TRUE  )

flow_images_from_data(x=img1,y=NULL, 
                       generator=gen_images,
                       batch_size=9,
                       save_to_dir="D:/data_flow17/aug/",
                       save_prefix="aug",
                       save_format="jpeg")

With kind regards,

Harry

**Comment by jjallaire** [comment]
What is a little tricky about `flow_images_from_data` is that it's really intended to be passed to the `fit` function rather than used standalone (although it certainly can, see below). The function returns an iterator rather than the actual images, so you need to traverse the iterator using `reticulate::iter_next()` to actually yield the images (again, this is handled for you when you use the function with `fit()`. 

Here's a complete example:

```r
library(keras)

img_width<-150
img_height<-150

img<-image_load("vignettes/examples/deep_dream.jpg", target_size=c(img_width,img_height))
img1<-image_to_array(img)
dim(img1)<-c(1,dim(img1))
dim(img1)
gen_images <- image_data_generator(rotation_range = 20,
                                   width_shift_range = 0.30,
                                   height_shift_range = 0.30,
                                   horizontal_flip = TRUE )

images_dir <- tempfile()
dir.create(images_dir)


images_iter <- flow_images_from_data(
  x=img1, y=NULL,
  generator=gen_images,
  batch_size=9,
  save_to_dir=images_dir,
  save_prefix="aug",
  save_format="jpeg"
)

reticulate::iter_next(images_iter)

list.files(images_dir)
```
